docs/developer/modules.rst

```diff
@@ -0,0 +1,16 @@
+Modules
+-------
+
+.. toctree::
+
+   modules/geomesa-compute
```

Comment:
Yeah, I am still not entirely comfortable with duplicating text here and in the READMEs. I am not sure what is the best long term solution there.